### ISSUE TITLE
API call to create ticket fails after upgrade to 6.4

### ISSUE THREAD

**Issue by rtheys** [zammad/zammad#5400]
### Used Zammad Version

6.4

### Environment

- Installation method: package
- Operating system (if you're unsure: `cat /etc/os-release` ): Debian 12
- Database + version:  postgresql 15
- Elasticsearch version: 8.15.3
- Browser + version: Firefox 132

### Actual behaviour

We've created a custom form where users can report issues. This form uses the zammad API to create the tickets. This has worked well, but since the upgrade to Zammad 6.4, this throws the following error in the log:

```
I, [2024-11-08T16:10:24.613470#827540-383960]  INFO -- : Started POST "/api/v1/tickets" for 134.58.58.10 at 2024-11-08 16:10:24 +0100
I, [2024-11-08T16:10:24.621724#827540-383960]  INFO -- : Processing by TicketsController#create as HTML
I, [2024-11-08T16:10:24.621817#827540-383960]  INFO -- :   Parameters: {"group"=>"IT", "priority_id"=>2, "state"=>"new", "customer_id"=>158, "title"=>"test abc", "type"=>"email", "article"=>{"type_id"=>1, "subject"=>"test abc", "body"=>"[FILTERED]", "content_type"=>"text/html", "type"=>"email", "internal"=>false, "sender"=>"Customer", "origin_by_id"=>158, "from"=>"Rik Theys <rik.theys@esat.kuleuven.be>", "created_by"=>"rik.theys@esat.kuleuven.be", "cc"=>""}}
I, [2024-11-08T16:10:24.843385#827555-179880]  INFO -- : ProcessScheduledJobs running...
I, [2024-11-08T16:10:24.845397#827555-179880]  INFO -- : Running job thread for 'Check channels.' (Channel.fetch) status is: sleep
I, [2024-11-08T16:10:24.845466#827555-179880]  INFO -- : Running job thread for 'Process ticket escalations.' (Ticket.process_escalation) status is: sleep
I, [2024-11-08T16:10:24.845746#827555-179880]  INFO -- : Running job thread for 'Check 'Channel' streams.' (Channel.stream) status is: sleep
I, [2024-11-08T16:10:24.845806#827555-179880]  INFO -- : Running job thread for 'Generate 'Session' data.' (Sessions.jobs) status is: sleep
I, [2024-11-08T16:10:24.849031#827555-179880]  INFO -- : Running job thread for 'Execute planned jobs.' (Job.run) status is: sleep
E, [2024-11-08T16:10:24.853228#827540-383960] ERROR -- : Sending an email without a valid recipient is not possible. (Exceptions::InvalidAttribute)
app/models/ticket/article.rb:403:in `check_email_recipient_validity'
app/controllers/concerns/creates_ticket_articles.rb:126:in `article_create'
app/controllers/tickets_controller.rb:191:in `block in create'
app/models/transaction.rb:17:in `block in execute'
app/models/transaction.rb:14:in `execute'
app/models/transaction.rb:26:in `execute'
app/controllers/tickets_controller.rb:82:in `create'
app/controllers/application_controller/has_download.rb:17:in `block (4 levels) in <module:HasDownload>'
app/controllers/application_controller/has_download.rb:16:in `block (3 levels) in <module:HasDownload>'
app/controllers/application_controller/has_download.rb:15:in `block (2 levels) in <module:HasDownload>'
app/controllers/application_controller/handles_transitions.rb:16:in `handle_transaction'
I, [2024-11-08T16:10:24.856325#827540-383960]  INFO -- : Completed 422 Unprocessable Entity in 234ms (Views: 1.8ms | ActiveRecord: 63.5ms | Allocations: 47397)
```

It seems to fail validation on the email recipient after the upgrade.

### Expected behaviour

Ticket created as with previous version (6.3.1).

### Steps to reproduce the behaviour

Issue an API post request to the /api/v1/tickets endpoint with the parameters as specified in the output above.

### I'm sure this is a bug and no feature request or a general question.

no

**Comment by MrGeneration** [comment]
You cannot create email tickets without proper recipients.

--

Sorry, but this seems to be a technical question.
Please note that this repository is the wrong place for technical questions.
Instead, please refer the Zammad community over at: https://community.zammad.org .

In case this turns out the be a bug and no technical question, we'll transfer the relevant information to this repo!

If you require commercial grade support and are no hosted or support contract customer, you can check out our support contracts here: https://zammad.com/pricing#selfhosted
*If you're paying customer already, please consult your Zammad support for assistance! :-)*

**Comment by rtheys** [comment]
Hi,

Indeed, adding a "to" field to the json we submit fixes the issue.

In Zammad 6.3, this was not needed. So it's either a regression in 6.4, or a bug in 6.3 that was fixed...

Anyway, I will close this ticket.

Thanks for your help and time.

Regards,
Rik

**Comment by dvuckovic** [comment]
Correct, this was introduced with fix for #5308.

Ticket customer has to have a valid email address now in case of email tickets, otherwise the validation will fail. This prevents possible issues with sending emails later to a non-existing recipient.

As you already identified, it's also possible to provide an actual recipient via the optional `to` field. Another option is to use `cc`.
